src/main/java/org/springframework/integration/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -0,0 +1,545 @@
+
+	private static final Log log = LogFactory.getLog(KafkaMessageListenerContainer.class);
+
+	public static final Function<Map.Entry<Partition, ?>, Partition> keyFunction = Functions.getKeyFunction();
+
+	private final GetOffsetForPartitionFunction getOffset = new GetOffsetForPartitionFunction();
+
+	private final PartitionToLeaderFunction getLeader = new PartitionToLeaderFunction();
+
+	private final Function<Partition, Partition> passThru = Functions.getPassThru();
+
+	private final LaunchFetchTaskProcedure launchFetchTask = new LaunchFetchTaskProcedure();
+
+	private final Object lifecycleMonitor = new Object();
+
+	private final KafkaTemplate kafkaTemplate;
+
+	private Partition[] partitions;
+
+	private String[] topics;
+
+	public boolean autoStartup = true;
+
+	private Executor fetchTaskExecutor;
+
+	private Executor adminTaskExecutor = Executors.newSingleThreadExecutor();
+
+	private int concurrency = 1;
+
+	private volatile boolean running = false;
+
+	private int maxFetch = KafkaConsumerDefaults.FETCH_SIZE_INT;
+
+	private int queueSize = 1024;
+
+	private MessageListener messageListener;
+
+	private ErrorHandler errorHandler = new LoggingErrorHandler();
+
+	private volatile OffsetManager offsetManager;
+
+	private ConcurrentMap<Partition, Long> fetchOffsets;
+
+	private ConcurrentMessageListenerDispatcher messageDispatcher;
+
+	private final MutableMultimap<BrokerAddress, Partition> partitionsByBrokerMap = Multimaps.mutable.set.with();
+
+	public KafkaMessageListenerContainer(ConnectionFactory connectionFactory, Partition... partitions) {
+		Assert.notNull(connectionFactory, "A connection factory must be supplied");
+		Assert.notEmpty(partitions, "A list of partitions must be provided");
+		Assert.noNullElements(partitions, "The list of partitions cannot contain null elements");
+		this.kafkaTemplate = new KafkaTemplate(connectionFactory);
+		this.partitions = partitions;
+	}
+
+	public KafkaMessageListenerContainer(final ConnectionFactory connectionFactory, String... topics) {
+		Assert.notNull(connectionFactory, "A connection factory must be supplied");
+		Assert.notNull(topics, "A list of topics must be provided");
+		Assert.noNullElements(topics, "The list of topics cannot contain null elements");
+		this.kafkaTemplate = new KafkaTemplate(connectionFactory);
+		this.topics = topics;
+	}
+
+	private static Partition[] getPartitionsForTopics(final ConnectionFactory connectionFactory, String[] topics) {
+		MutableList<Partition> partitionList = flatCollect(topics, new GetPartitionsForTopic(connectionFactory));
+		return partitionList.toArray(new Partition[partitionList.size()]);
+	}
+
+	public OffsetManager getOffsetManager() {
+		return offsetManager;
+	}
+
+	public void setOffsetManager(OffsetManager offsetManager) {
+		this.offsetManager = offsetManager;
+	}
+
+	public MessageListener getMessageListener() {
+		return messageListener;
+	}
+
+	public void setMessageListener(MessageListener messageListener) {
```

Comment:
I guess, we should introduce in the future `MessageListenerAdapter` for any POJO method invocation.